Drak: Making the Drak Ramp Up in response to threats

Zitchas opened the pull request:
**Content (Artwork / Missions / Jobs)**

## Summary
- Splits Drak.txt into a ships and outfits files.
- Adds "Augmented" versions of the AMC, Draining Field, Anti-Missile Field, Turret, and the "Drak Ammunition" that they all use.

These Augmented versions do significantly more damage than their normal equivalents, but use ammunition.

All Drak NPCs have been given frugal. As such, these augmented weapons will only be used when someone manages to inflict significant damage on them, thus creating a "ramp up" effect. They are "gentle" when not really threatened, and lethal when threatened.

## Details
The damage amounts of the Ember Tear and its comparative effectiveness compared to the AMC and the Archons drew attention to the fact that - for a faction that supposedly has weapons that can destroy whole star clusters, the Archons seem almost wimpy.

My personal lore is that the Archons are deliberately pulling their punches so that lesser species can *learn* from their interactions with the Archons (namely, learn the futility of war); but that doesn't explain why they don't step up their game when confronted by a serious threat.

Through the use of the "frugal" personality, which restricts the AI from using ammo based weapons until the ship has droped to 1.5 of its current health (100% shields counts for 1, and 100% hull counts for 1, combined to make 2. So frugal should "unleash" the full weaponry when any combination of shield or hull damage has dropped the total health to 1.5).

So, for most players, Archons should be just as friendly and whatnot as they always have been. And when someone inflicts enough damage on it to really matter, it takes off the gloves to deliver some serious hurt.

## Save File
This save file can be used to play through the new mission content:
(save file still to come)

## PR Checklist
 - ~~[X] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified~~
 - ~~[X] I uploaded the necessary image, blend, and texture assets here: {{insert link to assets}}~~
 - ~~[X] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets: {{insert PR link}}~~

Arachi-Lover commented on data/drak/drak ships.txt:
```diff
@@ -0,0 +1,123 @@
+		"outfit space" 1300
+		"weapon capacity" 1125
+		"engine capacity" 200
+		"energy capacity" 10000
```
Given that once it's angry it'll be firing twice as many weapons, and that the AMC already removes more than half the Archon's total energy cap (5610 out of 10000), I'd add another 0 to its energy capacity so that we don't have to worry about it lacking the energy at a particular instant.

Arachi-Lover replied:
This comment had a duplicate because GitHub was acting weird and wouldn't let me add this to the review for a while. I deleted the duplicate (I think), just saying in case anyone got confused.

Zitchas replied:
Fixed. Thanks!

Arachi-Lover commented on data/drak/drak ships.txt:
```diff
@@ -0,0 +1,123 @@
+		"outfit space" 1300
+		"weapon capacity" 1125
+		"engine capacity" 200
+		"energy capacity" 10000
```
Given that once it's angry it'll be firing twice as many weapons, and that the AMC already removes more than half the Archon's total energy cap (5610 out of 10000), I'd add another 0 to its energy capacity so that we don't have to worry about it lacking the energy at a particular instant.

Arachi-Lover reviewed (commented):
I still think the current weapons should receive buffs as well, something like 3x the damage for the AMC and 2x for the others.

Also, I'll reiterate that it'd be in their character to give most Archons (the exceptions being Lifted Lorax and Sleeping Dragon for the sake of the Void Sprites deal) the `forbearing` personality. It just doesn't sit well with me that if some Shuttle shoots an Archon with a Blaster it'll go ballistic on the little Shuttle.

Zitchas commented on data/drak/drak outfits.txt:
```diff
@@ -155,6 +281,29 @@ outfit "Drak Antimatter Fragment"
+		"shield damage" 56100
+		"hull damage" 56100
+		"blast radius" 250
+		safe
```
Thanks.

Zitchas commented on data/drak/drak outfits.txt:
```diff
@@ -176,90 +326,11 @@ effect "antimatter dust"
+outfit "Drak Ammunition"
+	plural "Drak Ammunition"
+	category "Ammunition"
+	cost 25000
```
Yeah, I think trillions of credits is a bit excessive. Somewhere in the billions should be sufficient, I think. I can bump it back up if we really feel the need to, but even with that ammunition dropped to 25 credits each in value, a single archon is still going to pretty much make a pilot's combat career.

Arachi-Lover commented on data/drak/drak outfits.txt:
```diff
@@ -134,7 +228,39 @@ outfit "Drak Antimatter Cannon"
+		"firing heat" 1870
+		"blast radius" 250
+		"hit force" 1000
+		safe
```
Another duplicate `safe`, missed it before.

Zitchas replied:
Thanks. And then I found a second one, too. I think all the places I added safes were actually duplicates.

Ferociousfeind commented:
Does frugal work with `ammo "outfit name" 0`? Because I'm pretty sure we support requiring an outfit to fire, but consuming none of it when firing too.

HTMLPhoton commented:
This will make the best way to kill an archon to disarm with status effects first. Would it make sense for frugal to start firing it's special weapons before it gets disarmed?

Amazinite commented:
For the time being with a single step up in power dictated by when frugal gets tripped, I wouldn't worry about the use of status effects. If we really wanted to combat status effect usage then we'd either give the Archon significant resistances to status effects or have it use weapons that bleed status effects.

(One potential weapon that I've discussed in the past is a weapon with an extremely high firing heat. This firing heat level would never be reached under normal firing circumstances, so if it does get reached we know that the ship is being attacked by heat damage, and since firing heat uses up the heat, any ship with this weapon could be effectively immune to heat damage, assuming the fire rate is quick enough such that by the time the heat level reaches the firing heat level again, the weapon is ready to fire, although it may be more interesting to have a longer reload time on such a weapon so that the first time you nearly overheat the ship it uses the weapon but then it's a race against the clock to overheat it while the weapon is on cooldown.)

In the future it'll probably be worth making use of #4224 to create multiple steps up in power as the Archon gets weaker, or if the implementation of #4224 provide more control over when a weapon is used aside from just the health of the firing ship then we could also create steps up in power should the Archon be hit by heavy status effects.

Zitchas commented:
Yeah, #4224 looks really good for that, and I'll definitely keep it in mind for the future. Likewise, having something that uses heat as fuel would be an interesting concept.

Ferociousfeind commented:
Status effects were pretty much already the best way- use huge amounts of slowing to hold the archon still so damage can take it out without having to chase it around

Ion doesn't do much, heat isn't gonna have an effect either, disruption might save you a little time but I dunno, waste of good potential DPS in my opinion. You still need huge DPS to do anything to it.

tehhowch commented:
I don't really see the need for this approach; Archon stats are not exactly "completed content." There's no reason to give a ship with some placeholder content _additional_ placeholder content that references the original placeholder content. 
If we've added other content that accidentally prevents them from having their proper role, then just bump the numbers or fix the balance of that added content.

Zitchas commented:
I view this as taking them a step towards being completed content. Can't do that in its entirety without added mechanics and a knowing what the end state of the rest of the game looks like; but expanding their behavior to include "stepping up" their performance in response to serious threats while maintaining their current "not super effective" combat for typical threats adds to their interest.

Things that react to the player and change in response to what they do is, generally speaking, a good improvement that makes people feel more respected and influential in the world. It also makes the Archons and the Drak feel a bit more "worthy" of their reputation. Honestly, it's kind of silly how easy it is to swarm them currently.

Amazinite commented on data/drak/drak outfits.txt:
```diff
@@ -176,90 +324,11 @@ effect "antimatter dust"
+	cost 25
+	thumbnail "outfit/unknown"
+	"mass" 0
+	"Drak Ammunition capacity" -1
```
For future reference, keep attribute names lowercase.

Amazinite commented on data/drak/drak outfits.txt:
```diff
@@ -176,90 +324,11 @@ effect "antimatter dust"
-	turret "Drak Turret"
-	turret "Drak Distancer"
-	turret "Drak Distancer"
+outfit "Drak Ammunition"
```
There's no need for an ammunition outfit, let alone so much of it.
1) Ammunition has a chance to get dropped on death. With as much ammunition as you're giving Archons, they'll probably be spitting out a load of crates.
2) Ammunition eventually runs out. Now with how much ammo you've given Archons I think it would take over 12 days straight to run out of ammo, but it'd be better if we didn't have to give the Archon so much and it never ran out.

Instead of having an ammo outfit, you should be able to just give every "augmented" weapon `ammo "Jump Drive" 0`, or choose any other outfit the Archon has installed. Since the weapon requires ammo to fire it should still be affected by the frugal personality, and an ammo requirement of 0 means it never actually uses up the ammo.

Zitchas replied:
OK, made the change.

Amazinite replied:
Still need to scrub the capacities from the weapons.

Amazinite reviewed (changes requested):
We usually split up files because they're getting too big, so there's really no need to split up this file because even with the additions it'd still only be a few hundred lines.

Zitchas commented:
OK, as per Amazinite's comments, I have removed the ammo and instead they simply require the presence of a Jump Drive in order to work. I have also added `forbearing` to all the Archons except the ones guarding Nenia.

Arachi-Lover commented:
Since the ammo is not longer there to boost up the Archon's cost, should its base hull cost be increased? To make it more rewarding in terms of combat-rating if one does manage to kill one after this buff.

Zitchas commented:
I'm ambivalent about making it rewarding to do so. How many people in the galaxy even know the Archon's exist? Kill one, and the Quarg should hate you, maybe the Korath love you, and everyone else is "what's an Archon?"

Amazinite commented:
> I still think the current weapons should receive buffs as well, something like 3x the damage for the AMC and 2x for the others.

I think we can spare reworking the base strength until later. At the very least any buff to the base weapons should keep a weakened version of the weapons for the gascraft missions, since as I recall the values are such that those missions aren't *too* brutal and give you some wiggle room with the Archon's spawn location and being able to survive a single AMC fragment exploding on an asteroid near you.

> There's no reason to give a ship with some placeholder content *additional* placeholder content that references the original placeholder content.

While it is placeholder content on top of placeholder content, I'd say the important part is the addition of the "ramping up" behavior, where in the future this will be much more dynamic and not just be buffed versions of the base weapons, but for now this serves as a sign of things to come.

Zitchas commented:
Re: housekeeping changes

Isn't the date typically the date the file was started? If anything, wouldn't we just add a second line with this year to show that it was started in 2016, then expanded in 2022? After all, MZ didn't do anything in this file this year...

Amazinite commented:
Well you split the original file, hence the new date to reflect *these* **new** files, and it's MZ's game, so any copyright not otherwise claimed goes to MZ. (And I've always been of the position that unless a file contains majority your own work *and* that work is significant, then the file header should point to MZ, as there's no point in adding a new line for every new person that touches a file, however small the change, and adding your name under the copyright for making a trivial change seems weird to me.)

Zitchas commented:
OK, that works. 

That being said, I've found it a little... odd.. that we'll attribute all but the most minor of updates to a sprite to the new artist, but for writers we have to be either the sole author or make massive changes to a file before having their name added to it. Not too important in this case, for obvious reasons, but as more of a general thing.
